app/buck2_execute_impl/src/executors/action_cache.rs

```diff
@@ -179,9 +176,10 @@ async fn query_action_cache_and_download_result(
         re_use_case,
         digest_config,
         manager,
+        &identity,
         buck2_data::CacheHit {
             action_digest: digest.to_string(),
-            action_key,
+            action_key: Some(identity.action_key.clone()),
```

Comment:
```suggestion
            action_key: if log_action_keys { Some(identity.action_key.clone()) } else { None },
```

To preserve existing behavior?

Reply:
Done, please see the updated PR